storage/aws/aws_test.go

```diff
@@ -0,0 +1,410 @@
+// skipped.  Run tests with `-parallel=1` to avoid concurent tests on the same
+// database, and specifically runs of `mustDropTables`.
+//
+// Sample command to start a local MySQL database using Docker:
+// $ docker run --name test-mysql -p 3306:3306 -e MYSQL_ROOT_PASSWORD=root -e MYSQL_DATABASE=test_tessera -d mysql
+package aws
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"database/sql"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"reflect"
+	"sync"
+	"testing"
+
+	gcs "cloud.google.com/go/storage"
+	"github.com/aws/smithy-go"
+	"github.com/google/go-cmp/cmp"
+	tessera "github.com/transparency-dev/trillian-tessera"
+	"github.com/transparency-dev/trillian-tessera/api"
+	"github.com/transparency-dev/trillian-tessera/api/layout"
+	storage "github.com/transparency-dev/trillian-tessera/storage/internal"
+	"k8s.io/klog"
+)
+
+var (
+	mySQLURI            = flag.String("mysql_uri", "root:root@tcp(localhost:3306)/test_tessera", "Connection string for a MySQL database")
+	isMySQLTestOptional = flag.Bool("is_mysql_test_optional", true, "Boolean value to control whether the MySQL test is optional")
+)
+
+// TestMain parses flags.
+func TestMain(m *testing.M) {
+	klog.InitFlags(nil)
+	flag.Parse()
+	os.Exit(m.Run())
+}
+
+// canSkipMySQLTest checks if the test MySQL db is available and if not, if the test can be skipped.
+//
+// Use this method before every MySQL test, and if it return true, exist the test.
+//
+// If is_mysql_test_optional is set to true and MySQL database cannot be opened or pinged,
+// the test will fail immediately. Otherwise, the test will be skipped if the test is optional
+// and the database is not available.
+func canSkipMySQLTest(t *testing.T, ctx context.Context) bool {
+	t.Helper()
+
+	db, err := sql.Open("mysql", *mySQLURI)
+	if err != nil {
+		if *isMySQLTestOptional {
+			return true
+		}
+		t.Fatalf("failed to open MySQL test db: %v", err)
+	}
+	defer func() {
+		if err := db.Close(); err != nil {
+			t.Fatalf("failed to close MySQL database: %v", err)
+		}
+	}()
+	if err := db.PingContext(ctx); err != nil {
+		if *isMySQLTestOptional {
+			return true
+		}
+		t.Fatalf("failed to ping MySQL test db: %v", err)
+	}
+	return false
+}
+
+// mustDropTables drops the `Seq`, `SeqCoord` and `IntCoord` tables.
+// Call this function before every MySQL test.
+func mustDropTables(t *testing.T, ctx context.Context) {
+	t.Helper()
+
+	db, err := sql.Open("mysql", *mySQLURI)
+	if err != nil {
+		t.Fatalf("failed to connect to db: %v", *mySQLURI)
+	}
+	defer func() {
+		if err := db.Close(); err != nil {
+			t.Fatalf("failed to close db: %v", err)
+		}
+	}()
+
+	if _, err := db.ExecContext(ctx, "DROP TABLE IF EXISTS `Seq`, `SeqCoord`, `IntCoord`"); err != nil {
+		t.Fatalf("failed to drop all tables: %v", err)
+	}
+}
+
+func TestMySQLSequencerAssignEntries(t *testing.T) {
+	ctx := context.Background()
+	if canSkipMySQLTest(t, ctx) {
+		klog.Warningf("MySQL not available, skipping %s", t.Name())
+		return
```

Comment:
Could you use `t.Skip("MySQL not available)` here (& elsewhere)?

Reply:
Ah there is it! I was looking for t.Return.. Thanks, I've used this, and left the warning messages. Otherwise one needs to specify -v to realize that these tests are not running.